dupserver.go

```diff
@@ -0,0 +1,56 @@
+package modgearman
+
+import (
+	"github.com/appscode/g2/client"
+)
+
+var dupjobsToSendPerServer = make(map[string](chan *answer))
+
+func initialiseDupServerConsumers(config *configurationStruct) {
+	if len(config.dupserver) > 0 {
+		for _, dupAddress := range config.dupserver {
+			logger.Debugf("creating dupserverConsumer for: %s", dupAddress)
+			dupjobsToSendPerServer[dupAddress] = make(chan *answer, config.dupServerBacklogQueueSize)
+			go runDupServerConsumer(dupAddress, dupjobsToSendPerServer[dupAddress], config)
+		}
+	}
+}
+
+func runDupServerConsumer(dupAddress string, channel chan *answer, config *configurationStruct) {
+	for {
+		item := <-channel
+		var error = sendResultDup(item, dupAddress, config)
+		if error != nil {
```

Comment:
this should be put in a retry loop. Otherwise it each item would be tried once and then discarded. I guess a loop which simply tries to send the message forever would be sufficient. Might add some 5 second sleep on errors as well. 5 Seconds do not hurt if there is a timeout anyway but would avoid a cpu killing loop if there is ex. a connection refused with an immediate error.